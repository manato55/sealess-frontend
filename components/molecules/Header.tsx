import React from 'react'
import Logo from '../atoms/Logo'
import ButtonToggle from '../organisms/ButtonToggle'
import styled from 'styled-components'
import { useRecoilValue } from 'recoil'
import { userStatus } from '../../store/atom'

interface Props {
    
}

export const Header = (props: Props) => {
    const user = useRecoilValue(userStatus)

    return (
        <>
            <StyledHeader>
                {user?.id && 
                <ButtonToggle />
                }
                <Logo />
            </StyledHeader>
        </>
    )
}


const StyledHeader = styled.header`
    height: 60px;
    width: 100%;
    background: #1e90ff;
    display: flex;
`;


export default Header