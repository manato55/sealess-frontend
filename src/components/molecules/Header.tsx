import React from 'react';
import Logo from '../atoms/Logo';
import ButtonToggle from '../organisms/ButtonToggle';
import styled from 'styled-components';
import { useRecoilValue } from 'recoil';
import { userStatus } from '../../store/atom';

interface Props {}

export const Header = (props: Props): React.ReactElement => {
  const user = useRecoilValue(userStatus);

  return (
    <>
      <StyledHeader>
        {user?.id && <ButtonToggle />}
        <MyInfo>
          {user?.department?.name}
          {user?.section?.name}&nbsp;{user?.name}
        </MyInfo>
        <Logo />
      </StyledHeader>
    </>
  );
};

const StyledHeader = styled.header`
  height: 60px;
  width: 100%;
  background: #1e90ff;
  display: flex;
`;

const MyInfo = styled.div`
  flex: 5;
  text-align: right;
  margin: auto 0;
  font-size: 15px;
`;

export default Header;
