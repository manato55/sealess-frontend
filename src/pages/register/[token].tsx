import RegisterToken from '../../components/organisms/RegisterToken';

const Token = () => {
  return (
    <>
      <RegisterToken />
    </>
  );
};

export default Token;
