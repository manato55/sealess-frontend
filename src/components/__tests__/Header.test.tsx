import React from 'react';
import Header from '../molecules/Header';
import { RecoilRoot } from 'recoil';
import { mount, shallow } from 'enzyme';
import { render, fireEvent } from '@testing-library/react';
import 'jsdom-global/register';
import { renderHook, act } from '@testing-library/react-hooks';
import renderer from 'react-test-renderer';
import { withHooks } from 'jest-react-hooks-shallow';

describe('header test', () => {
  test('snapshot', () => {
    const tree = renderer
      .create(
        <RecoilRoot>
          <Header />
        </RecoilRoot>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
