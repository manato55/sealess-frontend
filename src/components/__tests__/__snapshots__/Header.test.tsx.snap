// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header test snapshot 1`] = `
<header
  className="sc-hKFxyN ftQINu"
>
  <div
    className="sc-bdnxRM fvAkiQ"
  >
    <span>
      app
    </span>
  </div>
</header>
`;
